orm_lib/inc/drogon/orm/DbClient.h

```diff
@@ -114,10 +192,29 @@ class DbClient : public trantor::NonCopyable
         binder >> std::forward<FUNCTION2>(exceptCallback);
     }
 
+    /// JS style future
+    template <typename... Arguments>
+    std::shared_ptr<
+        JSStyleFuture<void(const Result &), void(const DrogonDbException &)>>
+    execSqlFuture(const std::string &sql, Arguments &&...args) noexcept
+    {
+        using Fut = JSStyleFuture<void(const Result &),
+                                  void(const DrogonDbException &)>;
+        std::shared_ptr<Fut> fut = std::make_shared<Fut>();
+        execSqlAsync(
+            sql,
+            [fut](const drogon::orm::Result &e) { fut->__success_handler(e); },
```

Comment:
You capture the fut here, so you don't need to capture itself in it's constructor.